lib/storage/src/content_manager/snapshots/recover.rs

```diff
@@ -91,7 +91,7 @@ async fn _do_recover_from_snapshot(
 
     log::debug!("Snapshot downloaded to {}", snapshot_path.display());
 
-    let tmp_collection_dir = Path::new(toc.storage_path())
+    let tmp_collection_dir = Path::new(toc.temp_path().unwrap_or_else(|| toc.storage_path()))
```

Comment:
Can we hide this logic into the `toc` and expose it through a function?

```rust
Path::new(toc.temp_path().unwrap_or_else(|| toc.storage_path()))
```

That's a bit of an implementation leak AND it is a repetition from the `toc` itself with

```rust
let temp_dir = Path::new(self.temp_path().unwrap_or_else(|| self.storage_path()))
            .join(SNAPSHOTS_TEMP_DIR);
```

WDYT?

Reply:
I think I'll add a `toc.temp_storage_path()` method for these.